hibernate-core/src/main/java/org/hibernate/type/descriptor/java/JavaType.java

```diff
@@ -192,7 +192,7 @@ default Comparator<T> getComparator() {
 		//noinspection unchecked
 		return Comparable.class.isAssignableFrom( getJavaTypeClass() )
 				? ComparableComparator.INSTANCE
-				: null;
+				: IncomparableComparator.INSTANCE;
```

Comment:
I assume the attribute that we could not compare (returning null) was the to-one `form` inside the `FormVersionId` embedded id, right? If that's the case, this way we're always ignoring associations inside identifiers when sorting actions - is this consistent with what we did in Hibernate 5?

I wonder if it would be possible to extract the associated entity's identifier, and then use that for comparing the associated values.